STM/src/main/java/org/jboss/stm/internal/proxy/LockManagerProxy.java

```diff
@@ -1035,21 +1035,21 @@ private boolean unpackObjectType (final Field afield, InputObjectState os)
             // TODO arrays
             
             if (afield.getType().equals(Boolean.class))
-                afield.set(_theObject, new Boolean(os.unpackBoolean()));
+                afield.set(_theObject, Boolean.valueOf(os.unpackBoolean()));
```

Comment:
You don't need to box the value before passing it to `afield.set`. So `afield.set(_theObject, os.unpackBoolean());` is sufficient. Similarly for the other changes in this routine and in other places where the same unnecessary boxing has been done in this PR.